### PR TITLE
ADEN-11141 Implementing UAP 4.0

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: src/ad-engine/utils/try-property.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { Dictionary } from '../models';
 
 export function whichProperty(obj: Dictionary = {}, properties: string[] = []): string | undefined {
+	if (!obj) {
```
Comment (Tomasz-at-Fandom) [inline src/ad-engine/utils/try-property.ts:4]: This smells hacky :) How is this related to any other changes in this PR?

Comment (dmnsobczak) [reply src/ad-engine/utils/try-property.ts:4]: Discussed on Slack, removing for now :)

**Review by Tomasz-at-Fandom — APPROVED** [review]
